[Flaky Test] HTTPJSON input - TestMetrics/Test_pagination_metrics - expected non zero value for metric httpjson_interval_pages_execution_time

## Flaky Test

* **Test Name:** TestMetrics/Test_pagination_metrics
* **Link:** https://github.com/elastic/beats/blob/18a7e5447a3b474f3f09f98932ff124d210067a5/x-pack/filebeat/input/httpjson/metrics_test.go#L162
* **Branch:** https://github.com/belimawr/beats/tree/log-raw-events-to-file-v2
* **Artifact Link:** https://buildkite.com/elastic/beats-xpack-filebeat/builds/2875#018f9bdc-bdc8-498f-b59a-20b57edaa95b/134-509
* **Notes:** 
  The test if flaky on Windows, running `go test -run=TestMetrics -count=10 -v ./input/httpjson` from `x-pack/filebeat`  was enough to consistently have the test failing at least once. It does not fail/reproduce on Linux, even with 3000 repetitions there was no failure. My [PR](https://github.com/elastic/beats/pull/38767) is only changing logging, I don't see how it could have affected the metrics from this input/this test.

### Stack Trace

```
2024-05-21 12:10:23 EDT	=== Failed
2024-05-21 12:10:23 EDT	=== FAIL: x-pack/filebeat/input/httpjson TestMetrics/Test_pagination_metrics (0.02s)
2024-05-21 12:10:23 EDT	    metrics_test.go:162:
2024-05-21 12:10:23 EDT	        	Error Trace:	C:/buildkite-agent/builds/bk-agent-prod-gcp-1716306902497376129/elastic/beats-xpack-filebeat/x-pack/filebeat/input/httpjson/metrics_test.go:162
2024-05-21 12:10:23 EDT	        	Error:      	Received unexpected error:
2024-05-21 12:10:23 EDT	        	            	expected non zero value for metric httpjson_interval_pages_execution_time
2024-05-21 12:10:23 EDT	        	Test:       	TestMetrics/Test_pagination_metrics
2024-05-21 12:10:23 EDT	
2024-05-21 12:10:23 EDT	=== FAIL: x-pack/filebeat/input/httpjson TestMetrics (0.02s)
```

It's also failing on `main` (`18a7e5447a3b474f3f09f98932ff124d210067a5`):
```
PS C:\devel\beats\x-pack\filebeat> go test -v -count=10 -run=TestMetrics .\input\httpjson\
go: downloading github.com/elastic/elastic-agent-libs v0.9.8
go: downloading github.com/elastic/elastic-agent-client/v7 v7.9.0
=== RUN   TestMetrics
=== RUN   TestMetrics/Test_pagination_metrics
--- PASS: TestMetrics (0.06s)
    --- PASS: TestMetrics/Test_pagination_metrics (0.06s)
=== RUN   TestMetrics
=== RUN   TestMetrics/Test_pagination_metrics
--- PASS: TestMetrics (0.03s)
    --- PASS: TestMetrics/Test_pagination_metrics (0.03s)
=== RUN   TestMetrics
=== RUN   TestMetrics/Test_pagination_metrics
--- PASS: TestMetrics (0.04s)
    --- PASS: TestMetrics/Test_pagination_metrics (0.04s)
=== RUN   TestMetrics
=== RUN   TestMetrics/Test_pagination_metrics
--- PASS: TestMetrics (0.06s)
    --- PASS: TestMetrics/Test_pagination_metrics (0.06s)
=== RUN   TestMetrics
=== RUN   TestMetrics/Test_pagination_metrics
    metrics_test.go:162:
                Error Trace:    C:/devel/beats/x-pack/filebeat/input/httpjson/metrics_test.go:162
                Error:          Received unexpected error:
                                expected non zero value for metric httpjson_interval_pages_execution_time
                Test:           TestMetrics/Test_pagination_metrics
--- FAIL: TestMetrics (0.05s)
    --- FAIL: TestMetrics/Test_pagination_metrics (0.05s)
=== RUN   TestMetrics
=== RUN   TestMetrics/Test_pagination_metrics
--- PASS: TestMetrics (0.05s)
    --- PASS: TestMetrics/Test_pagination_metrics (0.05s)
=== RUN   TestMetrics
=== RUN   TestMetrics/Test_pagination_metrics
--- PASS: TestMetrics (0.04s)
    --- PASS: TestMetrics/Test_pagination_metrics (0.04s)
=== RUN   TestMetrics
=== RUN   TestMetrics/Test_pagination_metrics
--- PASS: TestMetrics (0.05s)
    --- PASS: TestMetrics/Test_pagination_metrics (0.05s)
=== RUN   TestMetrics
=== RUN   TestMetrics/Test_pagination_metrics
--- PASS: TestMetrics (0.06s)
    --- PASS: TestMetrics/Test_pagination_metrics (0.06s)
=== RUN   TestMetrics
=== RUN   TestMetrics/Test_pagination_metrics
--- PASS: TestMetrics (0.03s)
    --- PASS: TestMetrics/Test_pagination_metrics (0.03s)
FAIL
FAIL    github.com/elastic/beats/v7/x-pack/filebeat/input/httpjson      0.551s
FAIL
PS C:\devel\beats\x-pack\filebeat>
```

--- a/CHANGELOG-developer.next.asciidoc
+++ b/CHANGELOG-developer.next.asciidoc
@@ -96,6 +96,7 @@ The list below covers the major changes between 7.0.0-rc2 and main only.
 - Cleaned up documentation errors & fixed a minor bug in Filebeat Azure blob storage input. {pull}36714[36714]
 - Fix copy arguments for strict aligned architectures. {pull}36976[36976]
 - Fix panic when more than 32767 pipeline clients are active. {issue}38197[38197] {pull}38556[38556]
+- Skip flakey metrics test on windows in filebeat httpjson input. {issue}39676[39676] {pull}39678[39678]
 
 ==== Added
 

--- a/x-pack/filebeat/input/httpjson/metrics_test.go
+++ b/x-pack/filebeat/input/httpjson/metrics_test.go
@@ -35,7 +35,7 @@ func TestMetrics(t *testing.T) {
 		skipReason string // GOOS:reason or GOOS,GOOS,...:reason.
 	}{
 		{
-			skipReason: "windows:flakey test on windows",
+			skipReason: "windows:flakey test on windows - see https://github.com/elastic/beats/issues/39676",
 
 			name: "Test pagination metrics",
 			setupServer: func(t *testing.T, h http.HandlerFunc, config map[string]interface{}) {

--- a/x-pack/filebeat/input/httpjson/metrics_test.go
+++ b/x-pack/filebeat/input/httpjson/metrics_test.go
@@ -8,6 +8,9 @@ import (
 	"fmt"
 	"net/http"
 	"net/http/httptest"
+	"runtime"
+	"slices"
+	"strings"
 	"testing"
 	"time"
 
@@ -28,8 +31,12 @@ func TestMetrics(t *testing.T) {
 		handler        http.HandlerFunc
 		expectedEvents []string
 		assertMetrics  func(reg *monitoring.Registry) error
+
+		skipReason string // GOOS:reason or GOOS,GOOS,...:reason.
 	}{
 		{
+			skipReason: "windows:flakey test on windows",
+
 			name: "Test pagination metrics",
 			setupServer: func(t *testing.T, h http.HandlerFunc, config map[string]interface{}) {
 				server := httptest.NewServer(h)
@@ -102,6 +109,9 @@ func TestMetrics(t *testing.T) {
 	for _, testCase := range testCases {
 		tc := testCase
 		t.Run(tc.name, func(t *testing.T) {
+			if reason := skipReason(tc.skipReason); reason != "" {
+				t.Skipf("skipping %s", reason)
+			}
 			tc.setupServer(t, tc.handler, tc.baseConfig)
 
 			cfg := conf.MustNewConfigFrom(tc.baseConfig)
@@ -162,4 +172,18 @@ func TestMetrics(t *testing.T) {
 			assert.NoError(t, tc.assertMetrics(reg))
 		})
 	}
+}
+
+func skipReason(s string) string {
+	if s == "" {
+		return ""
+	}
+	goos, reason, ok := strings.Cut(s, ":")
+	if !ok {
+		return s
+	}
+	if slices.Contains(strings.Split(goos, ","), runtime.GOOS) {
+		return reason
+	}
+	return ""
 }

